Source/WebCore/platform/mediastream/WebRTCProvider.h

```diff
@@ -0,0 +1,137 @@
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "MDNSRegisterError.h"
+#include "RTCDataChannelRemoteHandlerConnection.h"
+#include "RTCRtpCapabilities.h"
+#include "ScriptExecutionContextIdentifier.h"
+#include <wtf/CompletionHandler.h>
+#include <wtf/Expected.h>
+#include <wtf/UniqueRef.h>
+#include <wtf/text/WTFString.h>
+
+namespace WebCore {
+
+class ContentType;
+struct MediaCapabilitiesInfo;
+struct VideoConfiguration;
+struct MediaCapabilitiesDecodingInfo;
+struct MediaCapabilitiesEncodingInfo;
+struct MediaDecodingConfiguration;
+struct MediaEncodingConfiguration;
+
+class WebRTCProvider {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    WEBCORE_EXPORT static UniqueRef<WebRTCProvider> create();
+    WebRTCProvider() = default;
+    virtual ~WebRTCProvider() = default;
+
+    WEBCORE_EXPORT static bool webRTCAvailable();
+    WEBCORE_EXPORT static void setH264HardwareEncoderAllowed(bool);
+
+    virtual void setActive(bool);
+
+    using IPAddressOrError = Expected<String, MDNSRegisterError>;
+    using MDNSNameOrError = Expected<String, MDNSRegisterError>;
+
+    virtual RefPtr<RTCDataChannelRemoteHandlerConnection> createRTCDataChannelRemoteHandlerConnection() { return nullptr; }
+
+    using DecodingConfigurationCallback = Function<void(MediaCapabilitiesDecodingInfo&&)>;
+    using EncodingConfigurationCallback = Function<void(MediaCapabilitiesEncodingInfo&&)>;
+    void createDecodingConfiguration(MediaDecodingConfiguration&&, DecodingConfigurationCallback&&);
+    void createEncodingConfiguration(MediaEncodingConfiguration&&, EncodingConfigurationCallback&&);
+
+    WEBCORE_EXPORT void setH265Support(bool);
+    WEBCORE_EXPORT void setVP9Support(bool supportsVP9Profile0, bool supportsVP9Profile2);
+    WEBCORE_EXPORT void setVP9VTBSupport(bool);
+    bool isSupportingH265() const { return m_supportsH265; }
+    bool isSupportingVP9Profile0() const { return m_supportsVP9Profile0; }
+    bool isSupportingVP9Profile2() const { return m_supportsVP9Profile2; }
+    virtual bool isSupportingVP9VTB() const { return m_supportsVP9VTB; }
+    virtual void disableNonLocalhostConnections() { m_disableNonLocalhostConnections = true; }
+
+    bool isSupportingMDNS() const { return m_supportsMDNS; }
+
+#if ENABLE(WEB_RTC)
+    virtual std::optional<RTCRtpCapabilities> receiverCapabilities(const String& kind) = 0;
+    virtual std::optional<RTCRtpCapabilities> senderCapabilities(const String& kind) = 0;
+#endif
+
+    virtual void setLoggingLevel(WTFLogLevel) { }
+    virtual void setEnableWebRTCEncryption(bool) { }
+    virtual void setUseDTLS10(bool) { }
```

Comment:
I wonder whether these might not be libwebrtc specific

Reply:
I think the first might be doable in the GstWebRTC backend. I agree the two others are probably better off in the libwebrtc provider, I don't think those features are configurable in GstWebRTC.